internal/controller/promotions/promotions.go

```diff
@@ -665,3 +662,27 @@ func parseCreateActorAnnotation(promo *kargoapi.Promotion) string {
 	}
 	return creator
 }
+
+func calculateRequeueInterval(p *kargoapi.Promotion) time.Duration {
+	var (
+		defaultRequeueInterval = 5 * time.Minute
+		step                   = p.Spec.Steps[p.Status.CurrentStep]
+		runner                 = promotion.GetStepRunner(step.As)
+		timeout                = ptr.To(time.Duration(0))
+	)
+
+	if retryCfg, isRetryable := runner.(pkgPromotion.RetryableStepRunner); isRetryable {
+		timeout = retryCfg.DefaultTimeout()
+	}
+
+	var (
+		md            = p.Status.StepExecutionMetadata[p.Status.CurrentStep]
+		fiveMinLater  = time.Now().Add(defaultRequeueInterval)
```

Comment:
This seems like an awkward name for the variable, because if we ever change the value of `defaultRequeueInterval` we'll have to change the name of this variable as well.

Perhaps we don't need a variable for this at all (same for `targetTimeout`, perhaps) because the following seems both expressive and easy-enough to read:

`if md.StartedAt.Time.Add(*timeout).Before(time.Now().Add(defaultRequeueInterval)) {`

Reply:
That's fair.  Should we keep `targetTimeout` though because it's used both in the conditional and `time.Until(targetTimeout)`?